controllers/teleport_controller/teleport_controller.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2021 NUbots <nubots@nubots.net>
+ */
+
+#include <fstream>
+#include <iostream>
+#include <random>
+#include <webots/Supervisor.hpp>
+#include <array>
+
+#include "yaml-cpp/yaml.h"
+
+// AxisAngle rotations will be read from a config file and saved here
+std::vector<std::array<double, 4>> rotations;
+
+int main() {
+
+    // Load config file
+    try {
+        YAML::Node config = YAML::LoadFile("config.yaml");
+        rotations         = config["rotations"].as<std::vector<std::array<double, 4>>>();
+    }
+    catch (const YAML::BadFile& e) {
+        std::cerr << e.msg << std::endl;
+        return 1;
+    }
+    catch (const YAML::ParserException& e) {
+        std::cerr << e.msg << std::endl;
+        return 2;
+    }
+    
+
+    // create the Supervisor instance and assign it to a robot
+    webots::Supervisor supervisor = webots::Supervisor();
+    webots::Node& target          = *supervisor.getFromDef("nugus");
```

Comment:
This shouldn't apply to just one robot in the scene, it should be able to be applied to any robot and work.

Reply:
std::unique_ptr<Robot> nugus = std::make_unique<Robot>();